### PR TITLE
Martin molinero

### PR DESCRIPTION
**Pull request by MartinMolinero**

### CODE REVIEW

**Comment by MartinMolinero** [comment]
auth:
wizeline
Rocks!

File: src/components/App/App.test.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
```
Comment (dianasaurio) [inline src/components/App/App.test.jsx:2]: How could you test this using enzyme?

File: src/components/Menu/Menu.test.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { BrowserRouter } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
+import Menu from './Menu.component';
+import Adapter from 'enzyme-adapter-react-16';
+import { configure } from 'enzyme';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import videos from '../../utils/videoDataExample'
+
+configure({adapter: new Adapter()});
```
Comment (dianasaurio) [inline src/components/Menu/Menu.test.jsx:12]: You can create a setupTest.js file where you configure the enzyme adapter so you don't repeat the code in every test: 
https://create-react-app.dev/docs/running-tests/#initializing-test-environment

File: src/utils/test-utils.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// test-utils.js
+import React from 'react'
+import { render as rtlRender } from '@testing-library/react'
```
Comment (dianasaurio) [inline src/utils/test-utils.js:3]: Using enzyme, you don't need react testing library. Enzyme comes with methods to mount and shallow render a component: 
https://enzymejs.github.io/enzyme/docs/api/shallow.html

File: src/components/Private/Private.component.jsx [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
-  );
+const mapDispatchToProps = (dispatch) => {
+  return {
+  }
 }
 
-export default Private;
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps,
```
Comment (dianasaurio) [inline src/components/Private/Private.component.jsx:39]: If your mapDispatchToProps doesn't do anything, you can omit it or just pass a null

File: src/components/Searchbar/Searchbar.component.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        <Button id="search-button" onClick={this.setVideoQuery} variant="outline-info">
+          <SearchIcon />
+        </Button>
+      </Form>
+    );
+  }
+};
+
+const mapStateToProps = (state) => {
+  return state
```
Comment (dianasaurio) [inline src/components/Searchbar/Searchbar.component.jsx:46]: I see that you are connecting the component with the whole redux state but you aren't using it, if you are not using it, you can just pass the necessary values, in this case, the video search

File: src/components/VideoItem/VideoItem.test.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { render, screen } from '../../utils/test-utils'
+import { BrowserRouter } from 'react-router-dom';
+import VideoItem from './VideoItem.component';
+import Adapter from 'enzyme-adapter-react-16';
+import { configure } from 'enzyme';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import videos from '../../utils/videoDataExample'
+import userEvent from '@testing-library/user-event';
```
Comment (dianasaurio) [inline src/components/VideoItem/VideoItem.test.jsx:10]: Use enzyme capabilities like simulate function instead

File: src/pages/Explore/Explore.page.jsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        <Row>
+          <VideoList width={12} videos={this.props.videos.videos} />
+        </Row>
+      </Container>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  return state
```
Comment (dianasaurio) [inline src/pages/Explore/Explore.page.jsx:30]: If you are only connecting this component to prop drill to VideoList, I suggest VideoList be connected to the redux state instead, so you avoid unnecessary rendering for both components.

File: src/pages/Home/Home.page.authenticated.test.jsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    store = mockStore({
+      auth: {
+        id: 1,
+        avatarUrl: 'asdad',
+        name: 'Wizeline'
+      }
+    });
+    store.dispatch = jest.fn();
+    component = (
+      <Provider store={store}>
```
Comment (dianasaurio) [inline src/pages/Home/Home.page.authenticated.test.jsx:27]: Another way to test components using redux is to export the component without the connect HOC, so when you shallow render it in your tests, you can just pass the needed props.

File: src/pages/VideoDetail/VideoDetail.page.test.jsx [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+//             url: 'https://i.ytimg.com/vi/SiDYFCd9SH4/mqdefault.jpg',
+//             width: 320,
+//             height: 180,
+//           },
+//           high: {
+//             url: 'https://i.ytimg.com/vi/SiDYFCd9SH4/hqdefault.jpg',
+//             width: 480,
+//             height: 360,
+//           },
+//         },
```
Comment (dianasaurio) [inline src/pages/VideoDetail/VideoDetail.page.test.jsx:38]: Remove unused code

File: src/redux/reducers/videosReducer.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
+import { storage } from '../../utils/storage';
```
Comment (dianasaurio) [inline src/redux/reducers/videosReducer.js:1]: What does storages does? I don't understand why you call storage.set('videos', JSON.stringify(something)); on every case for the reducer

**Review by dianasaurio — CHANGES_REQUESTED** [review]

**Pull request closed without merging**